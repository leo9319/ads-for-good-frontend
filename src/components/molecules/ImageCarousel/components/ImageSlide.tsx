import React from 'react';
import { getStylesAsCssProperties } from '@utils/common/styles';
import Button from '@radix-styles/atoms/Button';
import { classNames } from '@utils/common/classNames';

import type Styles from '*.module.scss';

import { SlideProp } from '../ImageCarousel';

export interface ImageProps extends SlideProp {
  /**
   * Index of the slide
   */
  index: number;
  /**
   * CSS Module styles reference for the component
   */
  styles?: typeof Styles;
  /**
   * is it thumbnail view of the image or not
   */
  thumbnailMode?: boolean;
}

export const ImageSlide = ({
  src,
  index,
  tooltip,
  tooltipTitle,
  styles = {},
  thumbnailMode,
}: ImageProps) => {
  const getSlideClasses = (slide: string) =>
    getStylesAsCssProperties({ '--slide-image': `url(${slide})` });
  const thumbnail = thumbnailMode ? styles.thumbnailMode : null;
  const className = classNames(styles.slide, thumbnail);
  return (
    <>
      <div
        id={'slide-' + index}
        className={className}
        style={getSlideClasses(src)}
      />
      {tooltip && !thumbnailMode && (
        <div className={styles.tooltip}>
          <Button
            icon="icon-info"
            mode="inverse"
            size="sm"
            rounded
            ariaLabelName="Carousel Tooltip"
          />
          <div className={styles.tooltipText}>
            {tooltipTitle ? <h5>{tooltipTitle}</h5> : null}
            <p>{tooltip}</p>
          </div>
        </div>
      )}
    </>
  );
};

export default ImageSlide;
