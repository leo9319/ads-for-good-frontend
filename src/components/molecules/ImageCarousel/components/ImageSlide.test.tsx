import React from 'react';
import { render } from '@testing-library/react';
import ImageSlide from './ImageSlide';

describe('ImageSlide', () => {
  it('should render successfully', () => {
    const { baseElement } = render(
      <ImageSlide
        index={1}
        src="image.png"
        tooltip="image 1"
        tooltipTitle="1"
      />
    );
    expect(baseElement).toMatchSnapshot();
  });
});
