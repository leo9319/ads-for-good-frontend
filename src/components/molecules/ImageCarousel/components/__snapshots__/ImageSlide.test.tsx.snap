// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageSlide should render successfully 1`] = `
<body>
  <div>
    <div
      class=""
      id="slide-1"
      style="--slide-image: url(image.png);"
    />
    <div>
      <button
        aria-label="Carousel Tooltip"
        class="rt-reset rt-BaseButton rt-r-size-2 rt-variant-solid rt-Button wvButton sm-rounded inverse none undefined wvButtonRoundedIcon"
        data-accent-color=""
      >
        <div
          class="buttonContent iconLeft iconOnly"
        >
          <span
            class="icon"
            style="border-radius: 0;"
          >
            <span
              class="icon icon-info"
            />
          </span>
        </div>
      </button>
      <div>
        <h5>
          1
        </h5>
        <p>
          image 1
        </p>
      </div>
    </div>
  </div>
</body>
`;
